thylint/thylint.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+import re
+import json
+import argparse
+import sys
+
+DESCRIPTION = 'Check Isabelle theory files for unwanted outer syntax commands'
+
+# ignore content of parenthesised comments and Isabelle cartouches
+start_exp = re.compile(r'\(\*|\{\*|\\<open>')
+end_exp = re.compile(r'\*\)|\*\}|\\<close>')
+
+# ignore content of strings and inner syntax; we don't track cartouches/comments within strings
+string_delimiter = '"'
+
+
+def regexp_of(words):
+    """Turn word list into regexp.
+
+    Turn a list of word(ish) regexp strings into a compiled regexp that is the big union of
+    whole-word matches, with Isabelle-ish conventions for word boundaries. In particular,
+    count \' as a letter.
+    """
+    return re.compile('|'.join(r'\b(?<!\')(?:{0})\b(?!\')'.format(x) for x in words))
+
+
+diag_words = (
+    # title
+    "Interactive diagnostic command",
+    # message
+    "This command is usually used interactively only and should only be checked in for "
+    "demonstration purposes.",
+    # word(ish) regexps that should be flagged as warnings
```

Comment:
Out of curiosity, do you have a canonical source for this list, or is it 'just' a list of commands that you thought should be flagged.

The follow-up question is whether it's worth other people trying to think of more unwanted commands.

Reply:
I went through the output of `print_commands` and took everything that was interactive-only.